### PR TITLE
Add test logging for ExceptionsStoreTests

### PR DESCRIPTION
**Pull request by jander-msft**
###### Summary

Add logging to the ExceptionsStoreTests to help diagnose conditions when failures occur.

### CODE REVIEW

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsStoreTests.cs [hunk 1]
```diff
@@ -297,23 +307,36 @@ private sealed class ThresholdCallback : ExceptionsStoreCallbackBase
...
+                if (SetIfAtTarget(_addCompletionSource, ++_addCount, _addThreshold))
+                {
+                    _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.{nameof(AfterAdd)} threshold reached");
+                }
             }
 
             public override void AfterRemove(IExceptionInstance instance)
             {
-                SetIfAtTarget(_removeCompletionSource, ++_removeCount, _removeThreshold);
+                _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.{nameof(AfterRemove)}");
```
Comment (wiktork) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsStoreTests.cs:334]: Could [CallerMemberName] simplify some of these log entries

Comment (jander-msft) [reply src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsStoreTests.cs:334]: I don't think so. I'm not trying to capture the caller, I want to log that the current method is executing.

**Review by wiktork — APPROVED** [review]

**Pull request merged**